File: validParentheses.py
class Solution:
    # @return a boolean
    def isValid(self, s):

    	stack = []
        for char in s:
        	if char in ['(', '[', '{']:
        		stack.append(char)
        	else:
        		if len(stack) == 0:
        			return False
        		left = stack.pop()
        		if left == '{' and char == '}' or left == '(' and char == ')' or left == '[' and char == ']':
        			continue
        		else:
        			return False
        return len(stack) == 0

s = Solution()
print s.isValid("]]")

File: JAVA/validParentheses.java
public class Solution {
    public boolean isValid(String s) {
        Stack<Character> stk = new Stack<Character>();
        char last;
        for(char c : s.toCharArray()){
            if (c == '(' || c == '[' || c == '{')
                stk.push(c);
            else{
                if(stk.isEmpty())
                    return false;
                last = stk.pop();
                if(last == '(' && c == ')' || last == '[' && c == ']' || last == '{' && c == '}')
                    continue;
                return false;
            }
        }
        return stk.isEmpty();
    }
}